client/state/current-user/gravatar-status/reducer.js

```diff
@@ -0,0 +1,34 @@
+/**
+ * External dependencies
+ */
+import { combineReducers } from 'redux';
+
+/**
+ * Internal dependencies
+ */
+import {
+	GRAVATAR_UPLOAD_RECEIVE,
+	GRAVATAR_UPLOAD_REQUEST,
+	GRAVATAR_UPLOAD_REQUEST_SUCCESS,
+	GRAVATAR_UPLOAD_REQUEST_FAILURE
+} from 'state/action-types';
+import { createReducer } from 'state/utils';
+
+export const isUploading = createReducer( false, {
+	[ GRAVATAR_UPLOAD_REQUEST ]: () => true,
+	[ GRAVATAR_UPLOAD_REQUEST_SUCCESS ]: () => false,
+	[ GRAVATAR_UPLOAD_REQUEST_FAILURE ]: () => false
+} );
+
+export const tempImage = createReducer( {}, {
+	[ GRAVATAR_UPLOAD_RECEIVE ]: ( state, action ) => {
+		return {
+			src: action.src
+		};
+	}
+} );
```

Comment:
Regarding `createReducer`, there's good conversation in #4805 discussing when, why, and how it's useful. Biggest benefit currently is simplifying control over serialization, though I've also come to like it's compactness.